### PR TITLE
Fix panic in pending_runtime_api with Aura consensus env (#814)

### PR DESCRIPTION
**Pull request by tgmichel**
Rel MOON-1909

### CODE REVIEW

**Review by 4meta5 — APPROVED** [review]
This changes the user error when the header cannot be unwrapped from:
```
for all runtime APIs
"Runtime api access error: {:?}", e
```
to 
```
Cannot get header for block {:?}
```

**Comment by tgmichel** [comment]
> This changes the user error when the header cannot be unwrapped from:
> 
> ```
> for all runtime APIs
> "Runtime api access error: {:?}", e
> ```
> 
> to
> 
> ```
> Cannot get header for block {:?}
> ```

Before we were just assuming the header could be unwrapped (wrong) and now handle that error specifically. Runtime api access error message still the same.

**Comment by 4meta5** [comment]
Thanks for explaining! As far as I understand 

> Before we were just assuming the header could be unwrapped (wrong) and now handle that error specifically. 

We handle header not being unwrapped by returning the client error `Cannot get header for block {:?}`

> Runtime api access error message still the same.

Now I think the runtime API access error is not emitted if the header cannot be unwrapped

**Review by librelois — APPROVED** [review]

File: ts-tests/tests/test-balance.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ describeWithFrontier("Frontier RPC (Balance)", (context) => {
 	});
 
 	step("balance to be updated after transfer", async function () {
+		await createAndFinalizeBlock(context.web3);
```
Comment (crystalin) [inline ts-tests/tests/test-balance.ts:15]: This should go after the timeout line.
Also why do we need to create a block for this ?

**Comment by 4meta5** [comment]
So this PR applies https://github.com/paritytech/frontier/pull/842 to our frontier fork 

I'm still not sure what this has to do with pallet-randomness::on_initialize being introduced in 1802...

**Comment by tgmichel** [comment]
> So this PR applies https://github.com/paritytech/frontier/pull/842 to our frontier fork

Not really, or not only. This PR avoids the unwrap thing, but also builds the pending block over parent instead of best. And that's the main issue:

`api.initialize_block(&best, &header)` VS. `api.initialize_block(&parent_hash, &header)`

**Comment by 4meta5** [comment]
interesting, when is best != parent_hash?

**Comment by tgmichel** [comment]
> when is best != parent_hash?

Always. `best` is the tip of the canonical chain, and in this case `parent` is `best`'s parent.

**Pull request merged**